Fix tox support and only test supported versions of Python

kurtmckee opened the pull request:
This patch:

* Fixes tox support so that all versions of Python can be tested in one command
* Drops support for Python 2.6, 3.3, and 3.4.
* Adds support for Python 3.6 and 3.7.
* Adds HTML coverage reporting so that it's easy to see what is and is not getting tested.

Note that the unit tests are currently failing on **all** Python versions. I will be working to resolve this in a separate merge request.

Closes #1
Fixes #36

claird commented:
This sounds wonderful, Kurt.  It's going to be at least a couple of days
before I have time to examine it.  I apologize for the delay.




Cameron Laird, vice president
We make computers work for people.


On Tue, May 7, 2019 at 7:56 AM Kurt McKee <notifications@github.com> wrote:

> This patch:
>
>    - Fixes tox support so that all versions of Python can be tested in
>    one command
>    - Drops support for Python 2.6, 3.3, and 3.4.
>    - Adds support for Python 3.6 and 3.7.
>    - Adds HTML coverage reporting so that it's easy to see what is and is
>    not getting tested.
>
> Note that the unit tests are currently failing on *all* Python versions.
> I will be working to resolve this in a separate merge request.
>
> Closes #1 <https://github.com/claird/PyPDF4/issues/1>
> Fixes #36 <https://github.com/claird/PyPDF4/issues/36>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/claird/PyPDF4/pull/37
> Commit Summary
>
>    - Fix tox support and only test supported versions of Python
>
> File Changes
>
>    - *M* .gitignore
>    <https://github.com/claird/PyPDF4/pull/37/files#diff-0> (2)
>    - *M* README.md <https://github.com/claird/PyPDF4/pull/37/files#diff-1>
>    (6)
>    - *M* tox.ini <https://github.com/claird/PyPDF4/pull/37/files#diff-2>
>    (28)
>
> Patch Links:
>
>    - https://github.com/claird/PyPDF4/pull/37.patch
>    - https://github.com/claird/PyPDF4/pull/37.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/claird/PyPDF4/pull/37>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADM35HEUATMT62IC2RM43TPUGC75ANCNFSM4HLJC6UQ>
> .
>

kurtmckee commented:
Thanks, Cameron! I completely understand having limited time to review things. I hope your week is going well! =)

newnone commented:
@claird you can leave the review up to me. If it's ok, I can merge right away.

acsor committed:
Include indication on how to run the tests with unittest also

If tox isn't going to run more test code than unittest does, I propose to also leave the small tip of running with unittest. It just advertises a built-in option that's always there.

newnone commented:
I'd like to have a few days to go before the changes I have approved are integrated. I do not think such a slow pace is beneficial for the overall project.

newnone commented:
I merged manually this PR, minus my modification from [c4fbd73d9bf51d1629f1230414ad49aa711cdc03](https://github.com/claird/PyPDF4/pull/37/commits/c4fbd73d9bf51d1629f1230414ad49aa711cdc03). I wonder whether I can have GitHub recognize the merge automatically or close myself this ticket.
